src/modules/ctc/src/ctc.cc

```diff
@@ -143,14 +194,20 @@ std::vector<types::TrackCircuitData> Ctc::GetSuggestedSpeedsAndAuthorities(void)
 /*------------------------------------- Setters -------------------------------------*/
 void Ctc::SetBlocks(std::vector<types::Block> &blocks)
 {
-    types::Block yard;
-    yard.block = 0;
-    blocks_.push_back(yard);
-    for (types::Block &block : blocks)
+    if (blocks[CTC_FIRST_BLOCK].track == types::TrackId::TRACKID_GREEN)
     {
-        blocks_.push_back(block);
+        types::Block yard;
+        yard.block = 0;
+        yard.track = types::TrackId::TRACKID_GREEN;
+        blocks_.push_back(yard);
+        for (types::Block &block : blocks)
+        {
+            blocks_.push_back(block);
+        }
+        track_ = blocks[CTC_FIRST_BLOCK].track;
```

Comment:
You can refactor this method quite a bit to make it a lot simpler and clearer.

You can first set `track_` from the first block in the `blocks` vector, `emplace_back` a new block in `blocks_` for the yard block, set the track for the yard block equal to `track_`, and then use `insert` to append the `blocks` to `blocks_`.  No if/else statement or `for` loop necessary.

I also think it makes more sense to directly add the yard block to schedule and layout rather than add a new block for the yard along with whatever schedule or layout is imported.

Reply:
I agree we can put the yard block and let's do that in a different pr. For now I'll refactor the method to the suggestion